frontend/src/components/homepage/PreviewModal.tsx

```diff
@@ -118,37 +134,49 @@ const PreviewModal = ({
       <ModalOverlay />
       <ModalContent paddingLeft="20px">
         <ModalCloseButton position="static" marginLeft="auto" />
-        <ModalHeader paddingTop="-10px">
-          <Heading as="h3" size="lg" marginBottom="10px" width="75%">
-            {title}
-          </Heading>
-          <Badge variant="language" size="s">{`${language}`}</Badge>
-          <Badge
-            backgroundColor={getLevelVariant(level)}
-            size="s"
-          >{`Level ${level}`}</Badge>
-          {storyTranslationId && (
-            <Badge variant="stage" size="s">
-              {convertStageTitleCase(stage)}
-            </Badge>
+        {title &&
+          language &&
+          level &&
+          storyTranslationId &&
+          primaryBtnOnClick &&
+          primaryBtnText && (
```

Comment:
would checking for just `storyId` be sufficient to replace lines 137-142?

Reply:
or `storyTranslationId`

Reply:
Typescript gives me an error along the lines of `Argument of type 'number | undefined' is not assignable to parameter of type 'number'.` because the variables (`title, `language`, `level`, `storyTranslationId`, `primaryBtnOnClick`, `primaryBtnText`) are optional.

Reply:
Actually I can remove `storyTranslationId`, not sure why I had it in the first place.